ui/desktop/electron-app/src/ipc/handlers.js

```diff
@@ -116,6 +121,21 @@ handle('toggleFullscreenWindow', () => {
  */
 handle('closeWindow', () => app.quit());
 
+/**
+ * Toggle when signout is in progress
+ * to prevent app from quitting when signout modal is visible
+ */
+handle('setSignoutInProgress', (value) => (app.signoutInProgress = value));
+
+/**
+ * Toggle when attempt to close app is in progress
+ * and prevent closing app until user has engaged with modal
+ */
+handle('closeSessionsAndQuit', async () => {
```

Comment:
This should only need to just close the app since you already have a handler to stopAll